backtester/main.go

```diff
@@ -24,18 +26,96 @@ func main() {
 		fmt.Printf("Could not get working directory. Error: %v.\n", err)
 		os.Exit(1)
 	}
-	parseFlags(wd)
-	if !colourOutput {
+
+	flags := parseFlags(wd)
+	var btCfg *config.BacktesterConfig
+	if btConfigDir == "" {
+		btConfigDir = config.DefaultBTConfigDir
+		log.Infof(log.Global, "blank config received, using default path '%v'", btConfigDir)
+	}
+	fe := file.Exists(btConfigDir)
+	switch {
+	case fe:
+		btCfg, err = config.ReadBacktesterConfigFromPath(btConfigDir)
+		if err != nil {
+			fmt.Printf("Could not read config. Error: %v.\n", err)
+			os.Exit(1)
+		}
+	case !fe && btConfigDir == config.DefaultBTConfigDir:
+		btCfg, err = config.GenerateDefaultConfig()
+		if err != nil {
+			fmt.Printf("Could not generate config. Error: %v.\n", err)
+			os.Exit(1)
+		}
+		var btCfgJSON []byte
+		btCfgJSON, err = json.MarshalIndent(btCfg, "", " ")
+		if err != nil {
+			fmt.Printf("Could not generate config. Error: %v.\n", err)
+			os.Exit(1)
+		}
+		err = os.MkdirAll(config.DefaultBTDir, file.DefaultPermissionOctal)
+		if err != nil {
+			fmt.Printf("Could not generate config. Error: %v.\n", err)
+			os.Exit(1)
+		}
+		err = os.WriteFile(btConfigDir, btCfgJSON, file.DefaultPermissionOctal)
+		if err != nil {
+			fmt.Printf("Could not generate config. Error: %v.\n", err)
+			os.Exit(1)
+		}
+	default:
+		log.Errorf(log.Global, "non-standard config '%v' does not exist. Exiting...", btConfigDir)
```

Comment:
Have noticed an inconsistency with logging between starting with capitilised letters and then switching to lower case, would prefer all log messages to start with a capital similar to GCT.